includes/admin/class-wp-job-manager-admin.php

```diff
@@ -61,7 +61,41 @@ public function __construct() {
 	 * Set up actions during admin initialization.
 	 */
 	public function admin_init() {
+		global $wp_version;
+
 		include_once( dirname( __FILE__ ) . '/class-wp-job-manager-taxonomy-meta.php' );
+
+		if ( version_compare( $wp_version, JOB_MANAGER_MINIMUM_WP_VERSION, '<' ) ) {
+			add_action( 'admin_notices', array( $this, 'wp_version_admin_notice' ) );
+			add_filter( 'plugin_action_links_' . JOB_MANAGER_PLUGIN_BASENAME, array( $this, 'wp_version_plugin_action_notice' ) );
+		}
+	}
+
+	/**
+	 * Display notice if WordPress core is out-of-date in admin notice section.
+	 */
+	public function wp_version_admin_notice() {
+		// We only want to show the notices on the plugins page and WPJM admin pages.
+		$screen = get_current_screen();
+		$valid_screens = array( 'plugins', 'edit-job_listing', 'job_listing_page_job-manager-settings', 'edit-job_listing_type', 'edit-job_listing_category', 'job_listing' );
+		if ( null === $screen || ! in_array( $screen->id, $valid_screens ) ) {
+			return;
+		}
+
+		echo '<div class="error">';
+		echo '<p>' . sprintf( __( 'The upcoming release of <strong>WP Job Manager 1.31.0</strong> will require a more recent version of WordPress. <a href="%s">Please update WordPress</a> before updating WP Job Manager.', 'wp-job-manager' ), esc_url( self_admin_url( 'update-core.php' ) ) ) . '</p>';
```

Comment:
Once merged, I'll set up another PR that changes this to: 
echo '<p>' . sprintf( __( '<strong>WP Job Manager</strong> requires a more recent version of WordPress. <a href="%s">Please update</a> to the most recent version.', 'wp-job-manager' ), esc_url( self_admin_url( 'update-core.php' ) ) ) . '</p>';